library/core/src/pin.rs

```diff
@@ -485,6 +485,16 @@ impl<P: Deref<Target: Unpin>> Pin<P> {
     ///
     /// Unlike `Pin::new_unchecked`, this method is safe because the pointer
     /// `P` dereferences to an [`Unpin`] type, which cancels the pinning guarantees.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::pin::Pin;
+    ///
+    /// let val: u8 = 5;
+    /// // Wrap the value in a pin to make sure it doesn't move
```

Comment:
This isn't quite accurate. Since `Target: Unpin`, the value can move again.
```suggestion
    /// // The value doesn't care about being moved, so we can pin it just fine
```

I don't like my wording here a lot, feel free to improve it, but something like this is better.

Reply:
I guess that a better option is `Pin the value in memory to make sure that during the program lifecycle the value is not moved in another memory address`?